src/hash_table.js

```diff
@@ -0,0 +1,173 @@
+class HashEntry {
+  constructor(key, value) {
+    this.key = key
+    this.value = value
+    this.next = null
+  }
+}
+export default class HashTable {
+
+  constructor() {
+    this.currentSize = 0
+    this.storageMax = 128
+    this.storage = []
+  }
+
+  put(key, value) {
+    let hash = HashTable.hash(key) % this.storageMax
+
+    if(this.storage[hash]) {
+      this.storage[hash].insertNode(key, value)
+    } else {
+      this.storage[hash] = new Bucket(new HashEntry(key, value))
+    }
+
+    this.currentSize += 1
+
+    if(this.currentSize / this.storageMax >= 0.75) {
```

Comment:
I don't think you need to do this if you are using buckets to handle collisions. The Buckets should just grow in size.